src/database/mod.rs

```diff
@@ -55,6 +55,16 @@ pub struct SyncTime {
     pub block_time: BlockTime,
 }
 
+/// Structure encapsulates criteria used for deleting
+/// spent UTXOs
+#[derive(Clone, Debug)]
+pub struct DelCriteria {
+    /// Number of confirmations on UTXOs to make it eligible for deletion
+    pub confirmations: Option<u32>,
```

Comment:
This is a weird criteria IMHO... It would make sense to say "delete all the utxos spent more than N blocks ago", but that's not what you're doing: you're deleting the UTXOs that have been _received_ more than N blocks ago.
For example

```
Current time: block 15
Utxo A: reiceived at block 9, spent at block 10
Utxo B: received at block 1, spent at block 14
```

You probably want to delete utxo A, but keep  utxo B: it's been spent only since one block, so there could still be reorg, you might have to feebump it if it goes back in mempool, etc.

Reply:
Noted. I will change the implementation to "delete all the utxos spent more than N blocks ago".